annotations.md

```diff
@@ -22,4 +22,4 @@ This specification defines the following annotation keys, intended for but not l
 * **org.opencontainers.image.homepage** URL to find more information on the image (string, a URL with scheme HTTP or HTTPS)
 * **org.opencontainers.image.documentation** URL to get documentation on the image (string, a URL with scheme HTTP or HTTPS)
 * **org.opencontainers.image.source** URL to get source code for the binary files in the image (string, a URL with scheme HTTP or HTTPS)
-* **org.opencontainers.image.ref.name** Name of the reference (string)
+* **org.opencontainers.image.ref.name** Name of the reference for a target (string). Only valid when on descriptor within an index.json in [image layout](image-layout.md).
```

Comment:
+1 to splitting keys which should only appear in particular contexts out into a separate section. 

> This specification defines the following annotation keys, which MUST NOT be used in descriptors other than a certain descriptor.

This seems too strong for annotations (which are still described as “[arbitrary metadata][1]”.  Can we SHOULD this instead of MUSTing it, and say that the meaning of these keys are unspecified (or implementation-defined?) when used outside of their recommended location?

[1]: https://github.com/opencontainers/image-spec/blame/a8963c59859343b36b9a605239772aec70c945ef/descriptor.md#L44